android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt

```diff
@@ -599,6 +605,56 @@ class OkHttpTest {
   }
 
   @Test
+  fun testLoggingLevels() {
+    enableTls()
+
+    val testHandler = object : Handler() {
+      val calls = mutableMapOf<String, AtomicInteger>()
+
+      override fun publish(record: LogRecord) {
+        calls.getOrPut(record.loggerName) { AtomicInteger(0) }
+            .incrementAndGet()
+      }
+
+      override fun flush() {
+      }
+
+      override fun close() {
+      }
+    }.apply {
+      level = Level.FINEST
+    }
+
+    Logger.getLogger("")
+        .addHandler(testHandler)
+    Logger.getLogger("okhttp3")
+        .addHandler(testHandler)
+    Logger.getLogger(Http2::class.java.name)
+        .addHandler(testHandler)
+    Logger.getLogger(TaskRunner::class.java.name)
+        .addHandler(testHandler)
+    Logger.getLogger(OkHttpClient::class.java.name)
+        .addHandler(testHandler)
+
+    server.enqueue(MockResponse().setBody("abc"))
+
+    val request = Request.Builder()
+        .url(server.url("/"))
+        .build()
+
+    val response = client.newCall(request)
+        .execute()
+
+    response.use {
+      assertEquals(200, response.code)
+      assertEquals(Protocol.HTTP_2, response.protocol)
+    }
+
+    // Only logs to the test logger above
+    // Will fail if "adb shell setprop log.tag.okhttp.Http2 DEBUG" is called
+    assertEquals(setOf(OkHttpTest::class.java.name), testHandler.calls.keys)
```

Comment:
How did this get added?

Reply:
Ahhh it's in the test setup!

Reply:
#6011 drifted this I think. can someone with more experience confirm? Basically when this is invoked the level is always FINEST and hasn't to do with adb settings

Reply:
Probably, thanks for the attention here! Is it worth splitting this out into it's own test?

The goal is to make sure we don't accidentally enable debug logging for millions of Android apps filling mobile disks or causing PII incidents etc.

Reply:
> Probably, thanks for the attention here! Is it worth splitting this out into it's own test?

assume this is so that we don't inherit the logging stuff of the test rule? could make sense!

> The goal is to make sure we don't accidentally enable debug logging for millions of Android apps filling mobile disks or causing PII incidents etc.

this will be a good code comment to make over the test once we decide what to do with it